x/incentive/keeper/hooks_epoch.go

```diff
@@ -6,10 +6,22 @@ import (
 )
 
 // BeforeEpochStart performs a no-op
-func (k Keeper) BeforeEpochStart(_ sdk.Context, _ string, _ int64) {}
+func (k Keeper) BeforeEpochStart(ctx sdk.Context, epochIdentifier string, epochNumber int64) {
+
+}
 
 // AfterEpochEnd distributes vested tokens at the end of each epoch
 func (k Keeper) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, _ int64) {
+	// Find out incentive param using epochIdentifier and current block timestamp
+	foundIncentive, _, _ := k.FindProperIncentiveParm(ctx, epochIdentifier)
+
+	// If there is no incentive available with the current epoch and timestamp,
+	if !foundIncentive {
+		return
+	}
+
+	// Update uncommitted token amount
+	k.UpdateUncommittedTokens(ctx, epochIdentifier)
```

Comment:
Pass the incentives in here so they don't need to be retrieved again

Reply:
Yep, I couldn't do it. In order to do this, I should update parameter of IterateCommitment function inside commitment module to include incentive parameters. But we already imported commitment module from incentive module so I couldn't import inventive from commitment module.

Reply:
Fixed